llmfoundry/tokenizers/tiktoken.py

```diff
@@ -0,0 +1,276 @@
+                 pad_token: Optional[str] = None,
+                 **kwargs: Dict[str, Any]):
+        """Constructor creates a tiktoken tokenizer to use as the underlying.
+
+        tokenizer.
+
+        Args:
+            model_name (Optional[str], optional): The name of the model to load from tiktoken. Defaults to None.
+                Either model_name or encoding_name must be set, but not both.
+            encoding_name (Optional[str], optional): The name of the encoding to load from tiktoken. Defaults to None.
+                Either model_name or encoding_name must be set, but not both.
+            add_bos_token (bool, optional): Whether to add bos tokens. Defaults to False.
+            unk_token (Optional[str], optional): The unk token. Defaults to '<|endoftext|>'.
+            eos_token (Optional[str], optional): The eos token. Defaults to '<|endoftext|>'.
+            bos_token (Optional[str], optional): The bos token. Defaults to '<|endoftext|>'.
+            pad_token (Optional[str], optional): The pad token. Defaults to None.
+        """
+        try:
+            import tiktoken
+        except:
+            raise ImportError(
+                'You need to install tiktoken to use TiktokenTokenizerWrapper.')
+
+        if model_name is not None and encoding_name is not None:
+            raise ValueError(
+                'You need to specify either model_name or encoding_name, not both.'
+            )
+
+        self.model_name = model_name
+        self.encoding_name = encoding_name
+
+        if self.model_name is not None:
+            self.encoding = tiktoken.encoding_for_model(  # type: ignore (thirdParty)
+                self.model_name)
+        elif self.encoding_name is not None:
+            self.encoding = tiktoken.get_encoding(  # type: ignore (thirdParty)
+                self.encoding_name)
+        else:
+            raise ValueError(
+                'You need to specify either model_name or encoding_name.')
+
+        self.add_bos_token = add_bos_token
+
+        super().__init__(model_name=model_name,
+                         encoding_name=encoding_name,
+                         add_bos_token=add_bos_token,
+                         unk_token=unk_token,
+                         eos_token=eos_token,
+                         bos_token=bos_token,
+                         pad_token=pad_token,
+                         **kwargs)
+
+    @property
+    def vocab_size(self):
+        """Returns vocab size."""
+        return self.encoding.n_vocab
+
+    @property
+    def is_fast(self) -> bool:
+        return False
+
+    def get_vocab(self):
+        """Returns vocab as a dict."""
+        vocab = {}
+        for i in range(self.vocab_size):
+            try:
+                # need to try this first, so that we get a proper KeyError,
+                # otherwise it crashes in the rust code
+                _ = self.encoding.decode_single_token_bytes(i)
```

Comment:
is this standard?

Reply:
define "standard" it is how the tiktoken tokenizer functions.